ping/ping.c

```diff
@@ -82,6 +82,36 @@ ping_func_set_st ping4_func_set = {
 #define	NROUTES		9		/* number of record route slots */
 #define TOS_MAX		255		/* 8-bit TOS field */
 
+#define CASE_TYPE(x) case x: return #x;
+
+static char *str_family(int family)
+{
+	switch (family) {
+		CASE_TYPE(AF_UNSPEC)
+		CASE_TYPE(AF_INET)
+		CASE_TYPE(AF_INET6)
+	default:
+		error(2, 0, _("unknown protocol family: %d"), family);
```

Comment:
Ignorant question, but where does the memory come from for this?

Probably needs to be a static rotating buffer with at least 3 or 4 entries looking at the usage below.

Reply:
I'm missing something, are you talking about `error()` macro or `int family`?
`valgrind` is happy with this code.